src/main/java/com/minecolonies/coremod/colony/Colony.java

```diff
@@ -441,44 +506,273 @@ public int getID()
+                case TAG_FARMER_WHEAT:
+                    this.triggerAchievement(ModAchievements.achievementHarvest500Wheat);
+                    break;
+                case TAG_FARMER_POTATOES:
+                    this.triggerAchievement(ModAchievements.achievementHarvest500Potatoes);
+                    break;
+                case TAG_FARMER_CARROTS:
+                    this.triggerAchievement(ModAchievements.achievementHarvest500Carrots);
+                    break;
+                case TAG_LUMBERJACK_SAPLINGS:
+                    this.triggerAchievement(ModAchievements.achievementPlant500Saplings);
+                    break;
+                case TAG_LUMBERJACK_TREES:
+                    this.triggerAchievement(ModAchievements.achievementFell500Trees);
+                    break;
+                default:
+                    break;
+            }
+        }
+        if (statisticAmount >= NUM_ACHIEVEMENT_FIFTH)
+        {
+            switch (statistic)
+            {
+                case TAG_GUARD_MOBS:
+                    this.triggerAchievement(ModAchievements.achievementKill1000Mobs);
+                    break;
+                case TAG_MINER_ORES:
+                    this.triggerAchievement(ModAchievements.achievementMine1000Ores);
+                    break;
+                case TAG_MINER_DIAMONDS:
+                    this.triggerAchievement(ModAchievements.achievementMine1000Diamonds);
+                    break;
+                case TAG_BUILDER_HUTS:
+                    this.triggerAchievement(ModAchievements.achievementBuild1000Huts);
+                    break;
+                case TAG_FISHERMAN_FISH:
+                    this.triggerAchievement(ModAchievements.achievementCatch1000Fish);
+                    break;
+                case TAG_FARMER_WHEAT:
+                    this.triggerAchievement(ModAchievements.achievementHarvest1000Wheat);
+                    break;
+                case TAG_FARMER_POTATOES:
+                    this.triggerAchievement(ModAchievements.achievementHarvest1000Potatoes);
+                    break;
+                case TAG_FARMER_CARROTS:
+                    this.triggerAchievement(ModAchievements.achievementHarvest1000Carrots);
+                    break;
+                case TAG_LUMBERJACK_SAPLINGS:
+                    this.triggerAchievement(ModAchievements.achievementPlant1000Saplings);
+                    break;
+                case TAG_LUMBERJACK_TREES:
+                    this.triggerAchievement(ModAchievements.achievementFell1000Trees);
+                    break;
+                default:
+                    break;
+            }
         }
     }
 
     /**
-     * get the amount of killed mobs.
+     * Get the amount of statistic.
      *
-     * @return amount of mobs killed
+     * @param statistic the statistic.
+     * @return amount of statistic.
      */
-    public int getKilledMobs()
+    private int getStatisticAmount(@NotNull String statistic)
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Reduce the number of returns of this method 11, down to the maximum allowed 5. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS1142)